core/server/src/main/java/alluxio/worker/file/UnderFileSystemManager.java

```diff
@@ -191,24 +189,10 @@ private InputStream openAtPosition(long position) throws IOException {
         if (mStream != null) { // Close the existing stream if needed
           mStream.close();
         }
-        UnderFileSystem ufs = UnderFileSystem.get(mUri);
-        // TODO(calvin): Consider making openAtPosition part of the UFS API
-        if (ufs instanceof S3AUnderFileSystem) { // Optimization for S3A UFS
-          mStream =
-              new CountingInputStream(((S3AUnderFileSystem) ufs).openAtPosition(mUri, position));
-          mInitPos = position;
-        } else if (ufs instanceof S3UnderFileSystem) { // Optimization for S3 UFS
-          mStream =
-              new CountingInputStream(((S3UnderFileSystem) ufs).openAtPosition(mUri, position));
-          mInitPos = position;
-        } else if (ufs instanceof GCSUnderFileSystem) { // Optimization for GCS UFS
-          mStream =
-              new CountingInputStream(((GCSUnderFileSystem) ufs).openAtPosition(mUri, position));
-          mInitPos = position;
-        } else { // Other UFSs can skip efficiently, so open at start of the file
-          mStream = new CountingInputStream(ufs.open(mUri));
-          mInitPos = 0;
-        }
+        UnderFileSystem ufs = UnderFileSystem.Factory.get(mUri);
+        mStream =
+            new CountingInputStream(ufs.open(mUri, OpenOptions.defaults().setOffset(position)));
```

Comment:
This guarantees that the stream will be opened from the correct position?

Does that mean the block of code below is obsolete? @calvinjia